doc/src/parts/mathematical_foundations.rst

```diff
@@ -227,33 +240,72 @@ As it turns the method of least squares is a special case of the method
 
+Depending on the chosen cost function, uncertainties are handled differently.
+Simple pointwise gaussian uncertainties can be handled by a negative log likelihood cost function
+with a gaussian distribution for each data point.
+Alternatively the method of least squares can be used which can handle correlations between the
+uncertainties via the covariance or correlation matrix as well.
+The squared pointwise uncertainties correspond to the diagonal elements of the covariance matrix.
 
-Handling uncertainties
-======================
+Uncertainties relative to the data are constant, as the data is constant when performing a fit.
+With all standard cost functions it is only possible to respect uncertainties in *y*-direction.
+Respecting uncertainties in *x*-direction can be done by propagating the uncertainties onto the
+*y*-axis.
+*kafe2* does this by multiplying the derivatives of the model function at the data point location
+with the corresponding element of the given *x* covariance matrix.
 
-Gaussian uncertainties
-----------------------
+.. math::
+
+    \bm{V}_{ij} = \bm{V}_{y,ij} + f'\left(x_i\right)f'\left(x_j\right)\bm{V}_{x,ij}
+
+.. warning::
+    This only gives valid results, as long as the model function is approximately linear in the
+    range of the *x*-uncertainty.
+
+When using a covariance matrix *kafe2* is also capable of respecting uncertainties relative to the
+model as well.
+*x*-uncertainties depend on the model as well, and hence have to be updated after every step the
+function minimizer takes.
+This is handled internally by *kafe2*.
 
-Correlations
-------------
 
 Other types of uncertainties
-----------------------------
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+As the statistical uncertainty for histograms is a poisson distribution for the number of entries in
+each bin, a negative log-likelihood cost function with a poisson probability can be used, where
+:math:`{\bf d}` are the measurements or data points and :math:`{\bf m}` are the model predictions.
+
+.. math::
+
+    C = -2 \ln \prod_j \frac{{m_j}^{d_j}}{d_j!}\exp(-m_j)
 
+As the variance of a poisson distribution is determined by the mean value, no uncertainties have to
+be given when using poisson uncertainties.
 
-Cost functions
-==============
+Parameter constraints
+---------------------
 
-Least-squares ("chi-squared") estimator
----------------------------------------
+Parameter constraints are taken into account during the parameter estimation by adding an extra
+penalty to the cost function.
+The value of the penalty is determined by how well the parameter lies within its constraints.
+The new cost function then reads as
 
-:math:`\chi^2`
+.. math::
 
-Negative log-likelihood estimator
----------------------------------
+    C_\mathrm{total} = C + \mathrm{constraints}
 
+In general the penalty is given by the current parameter values :math:`\textbf{p}`, the expected
+parameter values :math:`\bm{\mu}` and the covariance matrix :math:`V`:
```

Comment:
This makes it sound like it's the same covariance matrix as used for the regular cost function.